### PR TITLE
fix some compile warnings

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Fix some compile warnings with elevated warning options.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: -
  - [ ] Backport for 3.10: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by MBkkt — APPROVED** [review]

File: arangod/Aql/Aggregator.cpp [hunk 1]
```diff
@@ -152,8 +152,6 @@ class MemoryBlockAllocator {
 struct AggregatorLength final : public Aggregator {
   explicit AggregatorLength(velocypack::Options const* opts)
       : Aggregator(opts), count(0) {}
-  AggregatorLength(velocypack::Options const* opts, uint64_t initialCount)
```
Comment (jsteemann) [inline arangod/Aql/Aggregator.cpp:155]: unused function

File: arangod/Aql/OptimizerRules.cpp [hunk 2]
```diff
@@ -288,11 +288,6 @@ class CollectionVariableTracker final
     return _dependencies[var];
   }
 
-  arangodb::aql::VarSet const& getCollectionVariables(
```
Comment (jsteemann) [inline arangod/Aql/OptimizerRules.cpp:291]: unused function

File: arangod/IResearch/ViewSnapshot.cpp [hunk 3]
```diff
@@ -43,8 +43,6 @@ namespace {
 class ViewSnapshotCookie final : public ViewSnapshot,
                                  public TransactionState::Cookie {
  public:
-  ViewSnapshotCookie() noexcept = default;
```
Comment (jsteemann) [inline arangod/IResearch/ViewSnapshot.cpp:46]: unused function

File: arangod/Replication2/Storage/WAL/LogReader.cpp [hunk 4]
```diff
@@ -172,9 +172,7 @@ auto LogReader::seekLogIndexBackward(LogIndex index)
     }
     _reader->seek(pos - sizeof(Record::Footer));
   }
-  return RT::error(TRI_ERROR_REPLICATION_REPLICATED_WAL_ERROR,
```
Comment (jsteemann) [inline arangod/Replication2/Storage/WAL/LogReader.cpp:175]: unreachable code

File: arangod/RestServer/DaemonFeature.cpp [hunk 5]
```diff
@@ -203,7 +203,7 @@ void DaemonFeature::checkPidFile() {
 
         try {
           oldPid = std::stol(oldPidS);
-        } catch (std::invalid_argument const& ex) {
```
Comment (jsteemann) [inline arangod/RestServer/DaemonFeature.cpp:206]: unused variable

File: lib/Basics/FloatingPoint.h [hunk 6]
```diff
@@ -274,11 +274,11 @@ class FloatingPoint {
 // We cannot use std::numeric_limits<T>::max() as it clashes with the max()
 // macro defined by <windows.h>.
 template<>
-inline float FloatingPoint<float>::Max() {
+[[maybe_unused]] inline float FloatingPoint<float>::Max() {
```
Comment (jsteemann) [inline lib/Basics/FloatingPoint.h:277]: unused function, but external code

File: lib/Basics/FloatingPoint.h [hunk 7]
```diff
@@ -274,11 +274,11 @@ class FloatingPoint {
...
   return FLT_MAX;
 }
 template<>
-inline double FloatingPoint<double>::Max() {
+[[maybe_unused]] inline double FloatingPoint<double>::Max() {
```
Comment (jsteemann) [inline lib/Basics/FloatingPoint.h:281]: unused function, but external code

**Pull request merged**